django_project/lesson/templates/worksheet/print.html

```diff
@@ -196,6 +196,14 @@ <h2>Further reading:</h2>
     </ul>
 </section>
 <footer>
+     <div class="col-lg-12" style="margin-bottom: 10px;
+     padding-left: 100pt;">
+            {% if worksheet.external_data %}
+                <p>Click <a href="{{ request.META.HTTP_HOST }}{{ worksheet.external_data.url }}">here</a> to
```

Comment:
DIsplay the full link instead of "click" in case the PDF is printed on paper. Also open it in a new tab.

Reply:
Will it not be too long @Gustry 
since the link holds the filepath.

Reply:
for instance
```
http://0.0.0.0:61202/media/images/lesson/worksheet/external_data/9792e82e2d69dd693ba79c405ae2323c21881284.dmp
```

Reply:
Because when the file gets stored it gets to have that long unique name hence when displaying that it will be very long.
That issue needs to be resolved in 
https://github.com/kartoza/projecta/issues/833

Reply:
Right.
But these links are at end of the PDF anyway, at the bottom, and the link is clickable, right? So I don't see it as a blocker. Nobody will read it if they don't need.

I know for a training, we will come with our USB sticks etc with data inside. But if somehow, the PDF is printed, and the student lost/doesn't have the data, at least links are useful he can still manage. The best would to create a short link, but this it out of scope for now.

Reply:
For openening in new tabs 
I have updated @Gustry